File: sdk/java/gladys-sensor-sdk/src/main/java/com/gladys/sensor/SensorRegistration.java
package com.gladys.sensor;

import gladys.v1.Orchestrator;

/**
 * One-shot registration helper for sensors.
 */
public final class SensorRegistration {

    private SensorRegistration() {
    }

    public static Orchestrator.RegisterResponse register(GladysClient client,
                                                          String sensorId, String sensorType,
                                                          Orchestrator.ComponentCapabilities capabilities) {
        return client.register(sensorId, sensorType, capabilities);
    }
}

File: sdk/js/gladys-sensor-sdk/src/SensorRegistration.ts
import { GladysClient } from "./GladysClient";
import {
  ComponentCapabilities,
  RegisterResponse,
} from "./generated/orchestrator";

/**
 * One-shot registration helper for sensors.
 */
export class SensorRegistration {
  static async register(
    client: GladysClient,
    sensorId: string,
    sensorType: string,
    capabilities: ComponentCapabilities
  ): Promise<RegisterResponse> {
    return client.register(sensorId, sensorType, capabilities);
  }
}
